### PR TITLE
route for creating restaurants added

### PR DESCRIPTION
**Pull request by ejubkadric94**
I will add the Admin authentication to the route as soon as the admin branch is verified and merged with the master.

### CODE REVIEW

File: app/controllers/RestaurantController.java [hunk 1]
```diff
@@ -80,4 +80,19 @@ public Result makeReservation(int id){
...
         return created(JsonSerializer.serializeObject(new ReservationHelper(reservation.getReservationId())));
     }
+
+    /**
+     * Creates a Restaurant and stores it.
+     *
+     * @return the restaurantId
+     */
+    public Result createRestaurant(){
+        response().setContentType("application/json");
```
Comment (nejra) [inline app/controllers/RestaurantController.java:90]: Check if content-type is implicitly set to application/json. If yes, remove this from all methods.

File: app/models/Restaurant.java [hunk 2]
```diff
@@ -11,10 +13,11 @@
  */
 @Entity
 @Table(name = "abh_restaurant")
-public class Restaurant extends Model{
+public class Restaurant extends Model implements Validation{
     @Id
     @Column(name = "restaurantId", columnDefinition = "BIGINT")
     @JsonView(RestaurantViews.BasicDetails.class)
```
Comment (nejra) [inline app/models/Restaurant.java]: Please, name classes in singular. In this case RestaurantView and BasicDetail. Also, do you really need RestaurantView class or you can have global classes like: BasicView, FullView or something similar?

File: app/utilities/PersistenceManager.java [hunk 3]
```diff
@@ -87,6 +88,23 @@ public static Token getToken(String token){
     }
 
     /**
+     * Persists restaurant into database.
+     *
+     * @param restaurant the Restaurant to be persisted
+     */
+    public static void saveRestaurant(Restaurant restaurant){
+        restaurant.getAddress().setRestaurantId(Restaurant.find.all().size() + 1);
```
Comment (nejra) [inline app/utilities/PersistenceManager.java]: This is a bad way of setting ID. ID should be generated via sequence on db side, not like this. This is not the practice and it doesn't support concurrency.    

@GeneratedValue(strategy = GenerationType.AUTO) is supported with auto_increment on db side. You should not worry about ID! Same for all models. You should never user setter for ID field.

File: app/utilities/PersistenceManager.java [hunk 4]
```diff
@@ -87,6 +88,23 @@ public static Token getToken(String token){
...
+        restaurant.getAddress().setEmail(restaurant.getName().replace(" ","").toLowerCase() + "@restauranttest.com");
```
Comment (nejra) [inline app/utilities/PersistenceManager.java]: Why is this hardcoded?

Comment (ejubkadric94) [inline app/utilities/PersistenceManager.java]: There was a "structure" issue regarding Address model class which forced me to add email to restaurants (which isn't needed at all). Now I solved that and fixed the hardcoded things. Won't happen again :)

File: app/utilities/PersistenceManager.java (hunk 3 above)
Comment (ejubkadric94) [inline app/utilities/PersistenceManager.java]: The thing I mentioned in previous comment was the source of that "complication", and of this one as well. Now it's all fixed.

File: app/models/Restaurant.java (hunk 2 above)
Comment (ejubkadric94) [inline app/models/Restaurant.java]: Now that is very logical and I have no clue why I thought that I'm supposed to create view class for each model. Also, I will make sure not to use plural class names again.

File: app/controllers/RestaurantController.java (hunk 1 above)
Comment (ejubkadric94) [inline app/controllers/RestaurantController.java:90]: Actually, the default response content-type is text/html, and I was looking for a way to change that, but the only way to do it was to set the content type to json in each Result of a controller. At least that is what play documentation says.

**Comment by nejra** [comment]
Can be merged :+1:

**Pull request merged**